docs/docs/architecture.md

```diff
@@ -1,143 +1,148 @@
 # Architecture overview
 
-If you are new to the concept of Confidential Computing, you may also find the [introduction to enclaves](enclaves.md)
-document helpful.
+This page covers the basic architecture of Conclave, a [Confidential Computing](enclaves.md) platform.
 
 ## Primary entities
 
-There are three entities in an application that uses Conclave:
+There are three main entities in any Conclave application.
 
-1. Clients
+1. Enclaves
 2. Hosts
-3. Enclaves
+3. Clients
 
-**Clients** send and receive encrypted messages to/from enclaves by interacting with the host over the network.
-Conclave doesn't mandate any particular network protocol for client<->host communication. It's up to you. However
-the content of the messages *is* defined, using the [Mail](#mail) API. Mail is described below.
+**Enclaves** are protected regions of memory in which you can write the confidential part of an application. An 
```

Comment:
Can you add that the enclave runs inside its own JVM, seperate from the host's JVM. Or is that mentioned later?

Reply:
Rewrote the para:
"Enclaves are protected regions of memory in which you can write the confidential part of an application. 
Although enclaves reside in a host program, a host or its owner cannot view or modify the code running in an enclave.
An enclave runs inside its own JVM, separate from the host's JVM."